#166789390 (ft) Get cars by using query parameters

willywax opened the pull request:
#### What does this PR do?
adds Endpoint to filter cars using query parameters
#### Description of Task to be completed?
add method to filter data. Returns result based on queries passed
#### How should this be manually tested?
By accessing the end point ***GET /cars*** and passing query parameters
#### Any background context you want to provide?
Queries that can be used are the instance variables of the Car class. 
state - [used,new], status-[available, sold] , min_price and max_price for price range, manufacturer as well as models
#### What are the relevant pivotal tracker stories?
[Delivers/Finishes #166789364](https://www.pivotaltracker.com/story/show/166789364)
[Delivers/Finishes ##166789390](https://www.pivotaltracker.com/story/show/166789390)

willywax committed:
ft(cars) Get car by userId

add get car from database using carId
add method to handle incorrect method

remove method to get car from array

[Starts #166789340 ]

willywax committed:
ft(cars) Filters data using query

add methods to filter data using [state,status,minprice, maxprice,manufacturer,model]

[Starts #166789364, #166789390] [Finishes #166789364, #166789390]

Kasulejoseph commented on controllers/cars.js:
```diff
@@ -14,42 +14,51 @@ exports.saveCar = (req, res, next) => {
+    } else {
+      let response =
+        cars.length === 0
+          ? new Response(200, cars, null, "No Car Found").response()
```
Not found shouldn't have a `200` status code, please update it with the right status code.

Kasulejoseph commented on controllers/cars.js:
```diff
@@ -14,42 +14,51 @@ exports.saveCar = (req, res, next) => {
+    .then(car => {
+      let response =
+        car.rows.length === 0
+          ? new Response(200, car.rows, null, "Car not Found").response()
```
Even here

willywax committed:
ft(cars) Search for car using filters

add response to return 404 status code when no car is returned

[Finishes/Delivers #166789364, #166789390, #166789340]
